### DISCUSSION TITLE
Multifile assemblies on non-Windows OSes

### DISCUSSION THREAD

**Discussion by sotlef** [dotnet/csharplang#1443] [Q&A] [answered]
It is hard for me to guess right project to address this issue, sorry if I misaddress it

Is it possible to create crossplatform multifile assembly according to documentation:  [Build a Multifile Assembly](https://github.com/dotnet/docs/blob/master/docs/framework/app-domains/how-to-build-a-multifile-assembly.md)

I mean possibility to:
- compile C++/CLI files to .netmodule on Windows (as I know there is no C++/CLI compiler on non-Windows OSes)
- compile C++ to native object files on linux for instance with gcc
- compile C# code to .netmodule
- link them all to one assembly with `al` with reference to `netstandard` or `netcore` frameworks?

**Comment by HaloFour** [comment, marked as answer]
I'd say that this probably belongs on either [CoreCLR](https://github.com/dotnet/coreclr) if the question is around the runtime support/behavior of such an assembly, or on [Roslyn](https://github.com/dotnet/roslyn) if the question is around the tooling of building such an assembly.
